pkg/services/policer/process.go

```diff
@@ -2,113 +2,80 @@ package policer
 
 import (
 	"context"
-	"sync"
+	"errors"
+	"time"
 
+	"github.com/nspcc-dev/neofs-node/pkg/local_object_storage/engine"
+	"github.com/nspcc-dev/neofs-sdk-go/object"
 	"go.uber.org/zap"
 )
 
-// Task represents group of Policer tact parameters.
-type Task struct{}
-
-type prevTask struct {
-	undone int
-
-	cancel context.CancelFunc
-
-	wait *sync.WaitGroup
-}
-
-type workScope struct {
-	val int
-
-	expRate int // in %
-}
-
 func (p *Policer) Run(ctx context.Context) {
-	defer func() {
-		p.log.Info("routine stopped")
-	}()
+	go p.poolCapacityWorker(ctx)
+	p.shardPolicyWorker(ctx)
+}
 
-	p.log.Info("process routine",
-		zap.Int("work scope value", p.workScope.val),
-		zap.Int("expansion rate (%)", p.workScope.val),
-		zap.Duration("head timeout", p.headTimeout),
+func (p *Policer) shardPolicyWorker(ctx context.Context) {
+	var (
+		addrs  []*object.Address
+		cursor *engine.Cursor
+		err    error
 	)
 
 	for {
-		select {
-		case <-ctx.Done():
-			p.prevTask.cancel()
-
-			p.log.Warn("context is done",
-				zap.String("error", ctx.Err().Error()),
-			)
-
-			return
-		case task, ok := <-p.trigger:
-			if !ok {
-				p.log.Warn("trigger channel is closed")
+		addrs, cursor, err = p.jobQueue.Select(cursor, p.batchSize)
+		if err != nil {
+			if errors.Is(err, engine.ErrEndOfListing) {
+				time.Sleep(1 * time.Second) // finished whole cycle, sleep a bit
+				continue
+			}
+			p.log.Warn("failure at object select for replication", zap.Error(err))
```

Comment:
```suggestion
			p.log.Warn("failure at object select for replication", zap.String("error", err.Error()))
```
and `\n` matters.

Reply:
Actually, `\n` are not produced anymore, even in wrapped errors! I think we simplified logger structure a while ago and it is not a case to worry now. 

```go
a := errors.New("deep error")
b := fmt.Errorf("included error: %w", a)
d := fmt.Errorf("included error two: %w", b)
c.log.Info("error", zap.Error(d)) // logger with config from neofs-dev-env

// 2021-11-24T09:50:09.841Z        info    neofs-node/main.go:59   error   {"error": "included error two: included error: deep error"} 
```

Reply:
Sorry, I meant `\n` is source code.

Reply:
Buy anyway, with current logger configuration, `zap.Error` is totally legit, though it produced some ugly output before and we stopped using it.